### PR TITLE
Fix vshpere

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Fix mismatch between kubermatic & machine-controller types

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/provider/datacenter.go [hunk 1]
```diff
@@ -29,15 +29,17 @@ type OpenstackSpec struct {
 	DNSServers []string `yaml:"dns_servers"`
 }
 
-// VSphereSpec describes a open stack datacenter
```
Comment (guusvw) [inline api/pkg/provider/datacenter.go:32]: =D nobody is doing c&p failure ;)

Comment (mrIncompetent) [reply api/pkg/provider/datacenter.go:32]: There are so many in our codebase :joy:

**Review by guusvw — APPROVED** [review]
LGTM

File: api/pkg/provider/cloud/vsphere/provider.go [hunk 2]
```diff
@@ -30,12 +31,12 @@ func (v *vsphere) getClient(cloud *kubermaticv1.CloudSpec) (*govmomi.Client, err
 		return nil, fmt.Errorf("invalid datacenter %q", cloud.DatacenterName)
 	}
 
-	u, err := url.Parse(dc.Spec.VSphere.Endpoint)
+	clientUrl, err := url.Parse(fmt.Sprintf("%s/sdk", dc.Spec.VSphere.Endpoint))
```
Comment (guusvw) [inline api/pkg/provider/cloud/vsphere/provider.go]: `clientURL` ;)

**Review by guusvw — CHANGES_REQUESTED** [review]

**Pull request merged**